### PR TITLE
Fix for pacts

### PR DESCRIPTION
**Pull request by kulla**
Include fixes for breaking pact tests:

* LicenseQuery shall return 404 when license cannot be found
* UserSetEmailQuery shall also return username

(There is still something broken, however it should be with the code in api / I'll have a look tomorrow)

### CODE REVIEW

File: server/src/user/model.rs [hunk 1]
```diff
@@ -187,17 +187,24 @@ impl User {
...
         payload: &user_set_email_mutation::Payload,
         executor: E,
-    ) -> Result<(), sqlx::Error>
+    ) -> Result<String, sqlx::Error>
     where
         E: Executor<'a>,
     {
+        let mut transaction = executor.begin().await?;
+
+        let username = sqlx::query!("select username from user where id = ?", payload.user_id)
```
Comment (hugotiburtino) [inline server/src/user/model.rs:196]: That does the job, that shouldn't be anyway complicated. My tendency would be to use a method of the uuid::model::user model.

**Comment by hugotiburtino** [comment]
The test for user_set_email is also failing.

File: server/src/user/model.rs (hunk 1 above)
Comment (kulla) [reply server/src/user/model.rs:196]: > That does the job, that shouldn't be anyway complicated. My tendency would be to use a fetch method of the uuid::model::user model. Do you think it is easy to use a fetch method instead?

can you try to implement it?

**Comment by kulla** [comment]
@hugotiburtino Ready for merge

**Review by hugotiburtino — APPROVED** [review]

File: server/src/user/model.rs (hunk 1 above)
Comment (hugotiburtino) [reply server/src/user/model.rs:196]: > can you try to implement it?
I'll try in another moment, putting in my TODO list.

Comment (hugotiburtino) [reply server/src/user/model.rs:196]: I've managed to do it locally, but it's not worthy, too much code for too little. Let's leave as it is.

**Pull request merged**